src/main/java/com/onevizion/scmdb/SqlScriptExecutor.java

```diff
@@ -186,4 +202,25 @@ private Connection getConnection(SchemaType schemaType, String schemaName) {
         }
     }
 
+    /**
+     * If there is a custom exit code 20001 in the context, a list of invalid objects found will be displayed
+     * and the application will end with code 1
+     * @param context ScriptRunnerContext after executed script
+     */
+    private void checkInvalidObjectOrExit(ScriptRunnerContext context) {
+        if (context != null && INVALID_OBJ_ERR_SQLCODE.equals(context.getProperty(ERR_SQLCODE))) {
+            String invalidObjMsg = String.valueOf(context.getProperty(ERR_MESSAGE_SQLCODE));
+            if (invalidObjMsg != null && !invalidObjMsg.isEmpty()) {
+                String invalidObjectMessage = invalidObjMsg.replaceAll(ORA_REGEXP, "");
+                System.err.print(invalidObjectMessage);
+                System.exit(EXIT_CODE_ERROR);
+            }
+        }
+    }
+
+    private int getExitCode(ScriptRunnerContext context) {
```

Comment:
working with `exitCode` is spread across classes. I propose to refactor `SqlScriptExecutor`. It should return `void` and throw exceptions if something fails, the exit code should be calculated inside `Scmdb` class

Reply:
Done

Reply:
Note: now catch() use `System.err.print` because `ColorLogger` reset logger configurate. We must create more powerfull logger for `Scmdb.class`